### ISSUE TITLE
requestAvatarId docstring should link to the credentials interfaces.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3346]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#3346|
|Type|enhancement|
|Created|2008-07-10 02:43:16Z|

The docstring for `requestAvatarId(credentials):` says: "something which implements one of the interfaces in self.credentialInterfaces." See http://twistedmatrix.com/documents/current/api/twisted.cred.checkers.ICredentialsChecker.html

It should link to the credentials module and `IUsernamePassword` to make things easier for people exploring the cred API.

<details><summary>Searchable metadata</summary>

```
trac-id__3346 3346
type__enhancement enhancement
reporter__jml jml
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1215657796000000 1215657796000000
changetime__1215775964000000 1215775964000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

```
Index: twisted/cred/checkers.py
===================================================================
--- twisted/cred/checkers.py    (revision 24246)
+++ twisted/cred/checkers.py    (working copy)
@@ -30,6 +30,8 @@
         avatar, an empty tuple to specify an authenticated anonymous user
         (provided as checkers.ANONYMOUS) or fire a Failure(UnauthorizedLogin).
         Alternatively, return the result itself.
+
+        @see: L{twisted.cred.credentials}
         """
 
 
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml set owner to @exarkun|
|-|-|

Thanks!

Merge approved.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [24259]) Add a link to the credentials module to the `ICredentialsChecker.requestAvatarId` docstring

Author: exarkun
Reviewer: jml
Fixes: [#3346](https://github.com/twisted/twisted/issues/3346)

The credentials object passed to `ICredentialsChecker.requestAvatarId` is an object providing
one or more of a number of interfaces; to try to help new users just learning cred, add a
link to the module defining most of Twisted's built in credentials interfaces to that
method's docstring.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
